### ISSUE TITLE
`mismatched-arg-count` false positive in example with `required-features`

### ISSUE THREAD

**Issue by Seldom-SE** [rust-lang/rust-analyzer#13416]
**rust-analyzer version**: 0.3.1238-standalone (61504c8d9 2022-10-08)

**rustc version**: 1.64.0 (a55dd71d5 2022-09-19)

I'm getting an error in the example for the following crate, even though it runs fine:

`Cargo.toml`
```toml
[package]
name = "bevy_test" # Not actually Bevy-related
version = "0.1.0"
edition = "2021"

[features]
feature = []

[[example]]
name = "example"
required-features = [ "feature" ]
```

`src/lib.rs`
```rust
#[derive(Default)]
pub struct Struct<T>(T);

impl Struct<Type1> {
    pub fn do_stuff(&self) {}
}

#[cfg(feature = "feature")]
impl Struct<Type2> {
    pub fn do_stuff(&self, _: i32) {}
}

#[derive(Default)]
pub struct Type1;

#[cfg(feature = "feature")]
#[derive(Default)]
pub struct Type2;
```

`examples/example.rs` (run with `cargo run --example example --features feature`)
```rust
use bevy_test::{Struct, Type2};

fn main() {
    Struct::<Type2>::default().do_stuff(0); // expected 0 arguments, found 1 rust-analyzer(mismatched-arg-count)
}
```

Thank you!

**Comment by Seldom-SE** [comment]
Error is also in rust-analyzer version: 0.4.1244-standalone (44fbe522e 2022-10-14)
